src/antsibull_docs/schemas/docs/plugin.py

```diff
@@ -207,6 +210,15 @@ class PluginDocSchema(BaseModel):
 
 class PluginExamplesSchema(BaseModel):
     examples: str = ""
+    examples_format: str = "yaml"
+
+    @p.validator("examples_format", always=True)
+    # pylint:disable=no-self-argument
+    def extract_examples_format(cls, value: t.Any, values: dict[str, t.Any]):
+        if isinstance(examples := values.get("examples"), str):
+            if fmt_match := _EXAMPLES_FMT_RE.match(examples.lstrip()):
+                value = fmt_match.group(1)
+        return value
```

Comment:
> Due to the line length limit this gets reformatted as [...] which IMO is less readable than two nested ifs.

Yeah, that's ugly. Black will leave this alone, though:

```suggestion
        # fmt: off
        if (
            isinstance(examples := values.get("examples"), str) and
            (fmt_match := _EXAMPLES_FMT_RE.match(examples.lstrip()))
        ):
        # fmt: on
            value = fmt_match.group(1)
        return value
```

Reply:
I dislike formatter directives even more than all the other solutions :) Do you mind just keeping the two `if`s?

Reply:
Alright, sure.